python/lsst/meas/algorithms/detection.py

```diff
@@ -439,31 +474,79 @@ def plotPeaks(fps, ctype):
+
+        Parameters
+        ----------
+        image : `afw.image.MaskedImage`
+            Image to measure noise statistics from if needed.
+        thresholdParity: `str`
+            One of "positive" or "negative", to set the kind of fluctuations
+            the Threshold will detect.
         """
         parity = False if thresholdParity == "negative" else True
-        threshold = afwDet.createThreshold(self.config.thresholdValue, self.config.thresholdType, parity)
+        threshold = afwDet.createThreshold(self.config.thresholdValue,
+                                           self.config.thresholdType, parity)
         threshold.setIncludeMultiplier(self.config.includeThresholdMultiplier)
 
         if self.config.thresholdType == 'stdev':
-            bad = image.getMask().getPlaneBitMask(['BAD', 'SAT', 'EDGE', 'NO_DATA', ])
+            bad = image.getMask().getPlaneBitMask(['BAD', 'SAT', 'EDGE',
+                                                   'NO_DATA', ])
             sctrl = afwMath.StatisticsControl()
             sctrl.setAndMask(bad)
             stats = afwMath.makeStatistics(image, afwMath.STDEVCLIP, sctrl)
-            thres = stats.getValue(afwMath.STDEVCLIP) * self.config.thresholdValue
+            thres = (stats.getValue(afwMath.STDEVCLIP) *
+                     self.config.thresholdValue)
             threshold = afwDet.createThreshold(thres, 'value', parity)
-            threshold.setIncludeMultiplier(self.config.includeThresholdMultiplier)
+            threshold.setIncludeMultiplier(
+                self.config.includeThresholdMultiplier
+            )
+
+        return threshold
+
+    def updatePeaks(self, fpSet, image, threshold):
+        """Update the Peaks in a FootprintSet by detecting new Footprints and
+        Peaks in an image and using the new Peaks instead of the old ones.
 
-        fpSet = afwDet.FootprintSet(image, threshold, maskName, self.config.minPixels)
-        return fpSet
+        Parameters
+        ----------
+        fpSet : `afw.detection.FootprintSet`
+            Set of Footprints whose Peaks should be updated.
+        image : `afw.image.MaskedImage`
+            Image to detect new Footprints and Peak in.
+        threshold : `afw.detection.Threshold`
+            Threshold object for detection.
+
+        Input Footprints with fewer Peaks than self.config.nPeaksMaxSimple
+        are not modified, and if no new Peaks are detected in an input
+        Footprint, the brightest original Peak in that Footprint is kept.
+        """
+        for footprint in fpSet.getFootprints():
+            oldPeaks = footprint.getPeaks()
+            if len(oldPeaks) <= self.config.nPeaksMaxSimple:
+                continue
+            # We detect a new FootprintSet within each non-simple Footprint's
+            # bbox to avoid a big O(N^2) comparison between the two sets of
+            # Footprints.
+            sub = image.Factory(image, footprint.getBBox())
```

Comment:
Yes.  We've actually always relied on that, because we run detection on a trimmed image (representing the area we could smooth) and then assume those Footprints and Peaks are valid for the untrimmedone.